pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -445,42 +421,47 @@ private PinotResourceManagerResponse uploadSegment(File indexDir, SegmentMetadat
         expectedCrc = Long.parseLong(expectedCrcStr);
       } catch (NumberFormatException e) {
         throw new ControllerApplicationException(LOGGER,
-            "Caught exception while parsing IF-MATCH CRC: " + expectedCrcStr, Response.Status.PRECONDITION_FAILED, e);
+            "Caught exception for segment: " + segmentName + " of table: " + offlineTableName
+                + " while parsing IF-MATCH CRC: " + expectedCrcStr, Response.Status.PRECONDITION_FAILED);
```

Comment:
Nit.
If expectedCrc is sent as a space character or somehow 0 length string, then we won't be able to see it in the log. Given that it has not parsed, it may be best to include the value of expectedCrcStr within single quotes

Reply:
Good point, added